lib/reek/source/source_code.rb

```diff
@@ -39,10 +40,10 @@ def initialize(code, origin, parser: Parser::Ruby22)
       # :reek:DuplicateMethodCall: { max_calls: 2 }
       def self.from(source)
         case source
-        when File     then new(source.read, source.path)
-        when IO       then new(source.readlines.join, IO_IDENTIFIER)
-        when Pathname then new(source.read, source.to_s)
-        when String   then new(source, STRING_IDENTIFIER)
+        when File       then new(source.read, source.path)
+        when IO         then new(source.readlines.join, IO_IDENTIFIER)
+        when SourcePath then new(source.read, source.to_s)
```

Comment:
> Does anyone use Examiner outside of the reek?

It's part of our public API, and yes, I know projects that use it directly (e.g., pronto-reek).